### PR TITLE
lxd provider: better errors when bootstrap failed

### PR DESCRIPTION
**Pull request by tych0**
When LXD isn't installed:

$ juju bootstrap bob lxd
ERROR invalid config: can't connect to the local LXD server: LXD socket not found; is LXD installed & running?

Please install LXD by running:
  $ sudo apt-get install lxd
and then configure it with:
  $ newgrp lxd
  $ lxd init

When the user doesn't have permissions:

$ juju bootstrap bob lxd
ERROR invalid config: can't connect to the local LXD server: Permisson denied, are you in the lxd group?

Please configure LXD by running:
  $ newgrp lxd
  $ lxd init

Signed-off-by: Tycho Andersen <tycho.andersen@canonical.com>

(Review request: http://reviews.vapour.ws/r/4424/)

### CODE REVIEW

File: tools/lxdclient/client_test.go [hunk 1]
```diff
@@ -30,7 +30,14 @@ func (cs *ConnectSuite) TestLocalConnectError(c *gc.C) {
...
 	// to the user.
-	c.Assert(err, gc.ErrorMatches, "can't connect to the local LXD server.*")
+	c.Assert(err.Error(), gc.Equals, `can't connect to the local LXD server: boo!
+
+Please install LXD by running:
+	$ sudo apt-get install lxd
+and then configure it with:
+	$ newgrp lxd
+	$ lxd init
+`)
```
Comment (jameinel) [inline tools/lxdclient/client_test.go]: Shouldn't there be a test for each of the 4 cases? (EACCES, ENOENT, ECONNREFUSED, and unknown error?

**Comment by jameinel** [comment]
The change looks very nice. Just want a couple tests to make sure we preserve the functionality.

Originally I thought we should install lxd, but I realize that requires sudo so we should probably ask the user to do it.

**Comment by tych0** [comment]
On Tue, Apr 05, 2016 at 07:11:39AM -0700, John Arbash Meinel wrote:
> The change looks very nice. Just want a couple tests to make sure we preserve the functionality.
> 
> Originally I thought we should install lxd, but I realize that requires sudo so we should probably ask the user to do it.

Well, and we want the user to configure their storage backend too
(i.e. hopefully use ZFS). Anyway, I'll add the test shortly.

**Comment by tych0** [comment]
Ok, I've added the tests. We needed one small patch to actually be able to test this code path, but other than that it turned out to be pretty simple.

**Comment by tych0** [comment]
Just a note that I rebased everything and added one more patch on top to warn if lxdbr0 is configured incorrectly (with a test).

**Comment by jameinel** [comment]
$$merge$$

**Pull request merged**